crates/matrix-sdk-crypto/src/identities/manager.rs

```diff
@@ -162,6 +162,27 @@ impl IdentityManager {
 
         self.store.save_changes(changes).await?;
 
+        // Update the sender data on any existing inbound group sessions based on the
+        // changes in this response.
+        //
+        // `update_sender_data_from_device_changes` relies on being able to look up the
+        // user identities from the store, so this has to happen *after* the
+        // changes from `handle_cross_signing_keys` are saved.
+        //
+        // FIXME: This thing is racy. If a session is created at the same time as a
+        //   `/keys/query` response is being processed, it could be saved
+        //   without up-to-date sender data, but it might be saved too late for
+        //   it to be picked up by `update_sender_data_from_device_changes`.
+        //
+        // FIXME: Come to that, what if this races against some other operation on the
+        //   session, such as marking it as needing backup? It's all a bit terrifying.
```

Comment:
Can we be any clearer what will happen if one of these races happens? I think we will lose one of the writes, so e.g. forget that a session needs backup, or forget that we trust the sender?